src/tests/simulation/src/kafka.rs

```diff
@@ -75,10 +75,19 @@ pub async fn producer(broker_addr: &str, datadir: String) {
             .expect("failed to create topic");
 
         let content = std::fs::read(file.path()).unwrap();
-        // binary message data, a file is a message
-        if topic.ends_with("bin") {
+        let msgs: Box<dyn Iterator<Item = &[u8]> + Send> = if topic.ends_with("bin") {
+            // binary message data, a file is a message
+            Box::new(std::iter::once(content.as_slice()))
+        } else {
+            Box::new(content.split(|&b| b == b'\n'))
```

Comment:
```suggestion
            Either::Left(std::iter::once(content.as_slice()))
        } else {
            Either::Right(content.split(|&b| b == b'\n'))
```

Reply:
This part is not on a critical path so I thought dynamic dispatch is acceptable. 🤪

Reply:
Just for omitting the `Box<dyn>` annotation. Both LGTM. 🤣